src/users/users.service.ts

```diff
@@ -11,16 +17,61 @@ export class UsersService {
     @InjectRepository(User) private userRepository: Repository<User>,
   ) {}
 
-  async create(user: CreateUserDto): Promise<User> {
+  /**
+   * its about semantics and good practises. therefore i decided to create an extra method for signing up
+   * @param createUserDto
+   */
+  signUp(createUserDto: CreateUserDto): Promise<User> {
+    return this.create(createUserDto);
+  }
+
+  /*
+   * register a user*/
+  private async create(createUserDto: CreateUserDto): Promise<User> {
+    const { username, password } = createUserDto;
+    try {
+      //const user = this.userRepository.create({ username, password });
+      return await this.userRepository.save({ username, password });
+    } catch (error) {
+      console.log(error.code);
+    }
+  }
+
+  /*
+   * later we may change the name of dto to authCredentialsDto or something more descriptive */
+  async signIn(createUserDto: CreateUserDto) {
+    const { username, password } = createUserDto;
+    const user = await this.userRepository.findOne({
+      where: {
+        username: username,
+      },
+    });
+
+    if (user && password === user.password) {
+      return user;
+    } else {
+      return null;
+      //throw new UnauthorizedException('Please check your login credentials');
+    }
+  }
+
+  async create2(user: CreateUserDto): Promise<User> {
     return await this.userRepository.save(user);
   }
 
   async findAll(): Promise<User[]> {
     return await this.userRepository.find();
   }
 
-  findOne(id: number) {
-    return `This action returns a #${id} user`;
+  async findOne(username: string, password: string) {
+    /* const user = await this.userRepository.findOne({where: username: username});
+
+    if (user && password === user.password) {
+      return user;
+    } else {
+      return null;
+      //throw new UnauthorizedException('Please check your login credentials');
+    }*/
```

Comment:
maybe before merging you could clean the code? There are a lot of grey areas (comments) like findOne method, or in private create function we have a line //const user = this.userRepository.create({ username, password }); which we don't need , also maybe we don't need create2 method? its a detail but I think it could make the code be more readable and less 'messy' :)